### PR TITLE
ES6 coding style adaptation on neo.blockchain.db.js

### PR DESCRIPTION
**Pull request by rockacola**

### CODE REVIEW

**Comment by hmcq6** [comment]
I see quite a few `let`s in this PR where the value of the variable is never reassigned. In JavaScript, when you store an object or array into a variable you're really storing a reference to that object or array. So long as that reference never changes its ok to use a `const` declaration.

For example:

```JavaScript
const five = 5;
five = 6; // Uncaught TypeError: Assignment to constant variable.

const groceryBasket = [];
groceryBasket.push('fish'); // No error as you are not reassigning the value in `groceryBasket`
```

With that in mind I think the only variables in this PR that should be declared `let` are `linkIndex` and `pointer`.

---

Another suggestion that I think might make this PR a little more ES6 inspired is if we used a destructuring assignment when requiring mongoose to extract only the parts we need.

```JavaScript
const { bSchema, model } = require('mongoose')
```

File: dist/neo.blockchain.db.js [hunk 1]
```diff
@@ -95,50 +96,52 @@ module.exports = function (network) {
...
+              )
+              /**
+               * If there is an asset list discripancy, scan for missing assets to update.
+               * This mechanic is used to automatically add new asset support as
+               * an asset it appears in a transaction.
+               */
+              if (res.assets.length !== node.assets.length) {
+                let included = _.map(res.assets, 'asset')
+                node.assets.forEach((asset) => {
+                  if (included.indexOf(asset.asset) === -1) {
```
Comment (hmcq6) [inline dist/neo.blockchain.db.js]: Could this be replaced with `Array.includes()`?

```JavaScript
if (!included.includes(asset.asset))
```

File: dist/neo.blockchain.db.js [hunk 2]
```diff
@@ -95,50 +96,52 @@ module.exports = function (network) {
...
+                (node.index - asset.index) >= blockAge
+              )
+              /**
+               * If there is an asset list discripancy, scan for missing assets to update.
+               * This mechanic is used to automatically add new asset support as
+               * an asset it appears in a transaction.
+               */
+              if (res.assets.length !== node.assets.length) {
+                let included = _.map(res.assets, 'asset')
+                node.assets.forEach((asset) => {
```
Comment (hmcq6) [inline dist/neo.blockchain.db.js]: If the only property we use from `asset` is `.asset` then maybe we should destructure the the first argument.

```JavaScript
node.assets.forEach(({ asset }) => {
  if (!included.includes(asset)) {
    parts[0].push({ asset })
  }
})
```

File: dist/neo.blockchain.db.js (hunk 1 above)
Comment (rockacola) [reply dist/neo.blockchain.db.js]: Suggestion is not related to ES6 adaptation. Separate code improvement PR recommended.

**Comment by lllwvlvwlll** [comment]
Are we OK to reject this? I think its been superceded by #60

**Pull request closed without merging**